biz.aQute.bndlib/src/aQute/bnd/build/WorkspaceExtensionHandler.java

```diff
@@ -0,0 +1,164 @@
+package aQute.bnd.build;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.osgi.resource.Capability;
+
+import aQute.bnd.build.Workspace.ResourceRepositoryStrategy;
+import aQute.bnd.header.Attrs;
+import aQute.bnd.header.Parameters;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Processor;
+import aQute.bnd.result.Result;
+import aQute.bnd.version.Version;
+import aQute.bnd.workspace.extension.WorkspaceExtensionNamespace;
+import aQute.lib.io.IO;
+
+public class WorkspaceExtensionHandler {
+
+	private Set<Capability>			availableWorkspaceExtensions	= new HashSet<>();
+	private Map<Capability, File>	enabledWorkspaceExtensions		= new HashMap<>();
+	private Workspace				ws;
+	private Processor				processor;
+
+	public WorkspaceExtensionHandler(Workspace ws) {
+		this.ws = ws;
+		processor = ws;
+	}
+
+	public WorkspaceExtensionHandler(Project project) {
```

Comment:
I don't understand how workspace extensions apply to projects. I can see you want to get some canned bnd files from a repository resource and apply it to the workspace. But how do Projects factor in here?

Reply:
As I initially wrote, the name workspace extensions is a bit misleading. In comparison to the current extension, the new version can extend the properties for the workspace, a project or a bndrun. A better name would be PropertiesExtension and is basically only a mechanism to include properties that are wrapped in a bundle.